lib/collection/src/collection_manager/holders/proxy_segment.rs

```diff
@@ -695,6 +695,33 @@ impl SegmentEntry for ProxySegment {
         self.write_segment.get().read().deleted_point_count()
     }
 
+    fn available_vectors_size_in_bytes(&self, vector_name: &str) -> OperationResult<usize> {
+        let wrapped_size = self
+            .wrapped_segment
+            .get()
+            .read()
+            .available_vectors_size_in_bytes(vector_name)?;
+        let wrapped_count = self.wrapped_segment.get().read().available_point_count();
+        let write_size = self
+            .write_segment
+            .get()
+            .read()
+            .available_vectors_size_in_bytes(vector_name)?;
+        let write_count = self.write_segment.get().read().available_point_count();
+        let deleted_points_count = self.deleted_points.read().len();
+        let stored_points = wrapped_count + write_count;
+        let avaliable_points = stored_points.saturating_sub(deleted_points_count);
+        // because we don't know the exact size of deleted vectors, we assume that they are the same avg size as the wrapped ones
+        if stored_points > 0 {
+            Ok(
+                ((wrapped_size as u128 + write_size as u128) * avaliable_points as u128
+                    / stored_points as u128) as usize,
+            )
```

Comment:
Another `saturating_add` and `saturating_mul` candidate?

Reply:
answered here: https://github.com/qdrant/qdrant/pull/4312#discussion_r1616029582
It's possible to get `u64` overflow and this overflow is a correct case, we need `u128` here